unity-client/Assets/Scripts/MainScripts/DCL/DataStore/Variables/BaseCollection.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Generic;
+
+public class BaseCollection<T> : IBaseCollection<T>, IEquatable<IEnumerable<T>>
+{
+    public event Action<IEnumerable<T>> OnChange;
+    public event Action<T> OnAdded;
+    public event Action<T> OnRemoved;
+
+    internal readonly List<T> list = new List<T>();
+
+    public BaseCollection() { }
+    public BaseCollection(IEnumerable<T> elements)
+    {
+        list = new List<T>(elements);
+    }
+
+
+    public IEnumerable<T> Get() => list;
+
+    public T this[int index]
+    {
+        get => list[index];
+        set => list[index] = value;
+    }
+
+    public void Set(IEnumerable<T> elements)
+    {
+        list.Clear();
+        list.AddRange(elements);
+
+        OnChange?.Invoke(list);
+    }
+
+    public void Add(T element)
+    {
+        list.Add(element);
+        OnAdded?.Invoke(element);
+    }
+
+    public bool Remove(T element)
+    {
+        if (!list.Remove(element))
+            return false;
+
+        OnRemoved?.Invoke(element);
+        return true;
+    }
+
+    public void RemoveAt(int index)
+    {
+        list.RemoveAt(index);
+    }
```

Comment:
question: shouldn't `RemoveAt` also raise `OnRemoved` event?

Reply:
Dammit, at the beggining was just a wrapper to other function that dissapeared. Gonna add it, nice catch!